### ISSUE TITLE
Multiline comments error

### ISSUE THREAD

**Issue by zerolx** [josdejong/mathjs#795]
Hi, recently I wrote some multi line expression in a app that I create and use mathjs, the expressions is similar a this:
_**That sample here is wrong made, do not generate errors**_
```
# comment
# comment
# any expression
1 + 1

```
_**That is the right one: (line break detail)**_
```
# comment


```
Then when executed is given the error:

> TypeError: Cannot read property 'compile' of undefined

if I put that expressions:
```
# comment
1 + 1
```
or that:
`1 + 1`
or that:
`# comment`
no error is given, but if i put this:
```
# comment
# anycomment
```
the same error appears again, it seems to be a bug in the code so I am reporting

I tested in latest version and happened the same

In console the error appoints to [mathjs/lib/expressions/function/eval.js#L44](https://github.com/josdejong/mathjs/blob/9101dafbad2fe68dbd3b4eb6e15e507bce06c438/lib/expression/function/eval.js#L44)
And is easy to reproduce

Best regards.

**Comment by josdejong** [comment]
Thanks for reporting, how can I reproduce your issue exactly? I tried:

```js
var text = '# comment\n# comment\n# any expression\n1 + 1';

math.eval(text) // works fine, returns a ResultSet [2]
```

**Comment by zerolx** [comment]
I made a fiddle for this
[https://jsfiddle.net/galba/x3b063wq/]()

And sorry, the first sample that I give (and you used) was wrong, that really do not generate error, but the other yes.
Try at the fiddle, I put samples there too

**Comment by josdejong** [comment]
Thanks a lot, I see the problem now, it can be simpified to:

```js
math.eval('\n')  // throws TypeError: Cannot read property 'compile' of undefined
```

**Comment by josdejong** [comment]
This has been fixed now in the `develop` branch and will be solved in the first next release.

**Comment by zerolx** [comment]
Thanks!

**Comment by josdejong** [comment]
This has been fixed in `v3.9.2`
